### PR TITLE
Use new ECL I/O classes in compareECL

### PR DESCRIPTION
**Pull request by akva2**
Split from #696, sits on top of #699 

Note: cannot currently be merged due to issues in opm-simulator tests.

### CODE REVIEW

File: examples/test_util/EclFile.cpp [hunk 1]
```diff
@@ -16,490 +16,807 @@
...
         }
 
-        for (int i=0;i<num;i++){
+        array_size.push_back(num);
+        array_type.push_back(arrType);
 
-            bool value;
-            unsigned int intVal;
+        array_name.push_back(EIOD::trimr(arrName));
+        array_index[array_name[n]] = n;
```
Comment (bska) [inline examples/test_util/EclFile.cpp]: Array_index is a `std::map<std::string, int>`.  Does this mean that if a name repeats&mdash;as would be the case for the header arrays (e.g., `INTEHEAD`) of a unified restart file&mdash;that `array_index` records only the last occurrence of the array?

Comment (tskille) [reply examples/test_util/EclFile.cpp]: Yes that is correct, and this is the one of the main reasons why we have to make specialized classes for unified restart files, summary files, and rft files. In the restart example below, your will get pressure from restart file SPE1CASE1 for report step number 12 (SEQNUM=12)

ERst rst1("SPE1CASE1.UNRST");
loadReportStepNumber(12);
rst1.getRst("PRESSURE", 12);

In program compareECL only INIT files are compared with the using EclFile.  Feel free to call me if you have other questions, or if you would like to discuss how this is implemented

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
Just a quick note: I had to make a number of changes of the form
```diff
 
 std::vector<EclFile::EclEntry> ERft::listOfRftArrays(const std::string& wellNam
e,
                                                      int year, int month, int d
ay) const
 {
-    return listOfRftArrays(wellName, {year, month, day});
+    return listOfRftArrays(wellName, RftDate{year, month, day});
 }
```
to get this PR to compile on my Linux Mint 17.2 (roughly Ubuntu 14.04.2 LTS) workstation using GCC 4.9.  The reason is that GCC 4.9 does not have a sufficiently well developed C++14 standard library implementation so the conditionally explicit tuple constructors are all explicitly marked `explicit`.

Do we care enough about Ubuntu 14.04 and GCC 4.8 (or 4.9) to add those workarounds?

**Comment by atgeirr** [comment]
> Do we care enough about Ubuntu 14.04 and GCC 4.8 (or 4.9) to add those workarounds?

I'd say no, as it takes extra time and Ubuntu 14.04 has not been supported (from OPM) for more than two years I think. I think you have to upgrade... If there are reasons why this is not possible I won't mind too much merging a workaround once, but it's bound to happen again and again and waste more time until you upgrade.

**Comment by bska** [comment]
> > Do we care enough about Ubuntu 14.04 and GCC 4.8 (or 4.9) to add those workarounds?
> 
> I'd say no, as it takes extra time and Ubuntu 14.04 has not been supported (from OPM) for more than two years I think.

Fair enough.  I'll run the local patch while it's needed.  To be honest, I'm more concerned about RHEL 7 whose system compiler is GCC 4.8 and would run into the same issue.  If we're not using the system compiler for packaging I'll swallow the cost for now.

**Comment by atgeirr** [comment]
> If we're not using the system compiler for packaging

@akva2 you can probably confirm this?

**Comment by akva2** [comment]
It is costless imo. Post your diff somehow

**Comment by akva2** [comment]
We use devtoolset-6 which is gcc 6.4 i believe


-------- Opprinnelig melding --------
Fra: Atgeirr Flø Rasmussen <notifications@github.com>
Dato: 05.04.2019 18:25 (GMT+01:00)
Til: OPM/opm-common <opm-common@noreply.github.com>
Ko: Arne Morten Kvarving <Arne.Morten.Kvarving@sintef.no>, Mention <mention@noreply.github.com>
Emne: Re: [OPM/opm-common] Use new ECL I/O classes in compareECL (#703)


If we're not using the system compiler for packaging

@akva2<https://github.com/akva2> you can probably confirm this?

**Comment by andlaus** [comment]
I agree that we should not care about such crufty compilers too much if nobody is willing to put sufficient resources behind it. Though, in my not so humble opinion, we should explicitly to decide to require c++-2014 or -- even better -- c++-2017: C++-2014 -- or rather a  compiler with a fully C++-2011 compliant standard library (i.e., GCC >= 5.0) -- would allow us to get rid of `boost::regex` in favor of `std::regex` and C++-2017 (GCC >= 8.0) would enable us to replace `boost::filesystem` by `std::filesystem`. I guess that's post release stuff, though.

**Comment by bska** [comment]
> Post your diff somehow

I've created pull request akva2/opm-common#6 to this effect.

**Comment by akva2** [comment]
jenkins build this please

**Comment by bska** [comment]
The [Jenkins test failures](https://ci.opm-project.org/job/opm-common-PR-builder/921/#showFailuresLink) here are interesting. They generally fall in two categories,

  1. Summary vectors of different size between base run and restarted runs ([SPE1CASE2_ACTNUM](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareRestartedSim_flow_SPE1CASE2_ACTNUM/), [SPE9_CP_SHORT](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareRestartedSim_flow_SPE9_CP_SHORT/), and [SPE1CASE2_ACTNUM, parallel](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareParallelRestartedSim_flow_SPE1CASE2_ACTNUM/))

  2. Differing set of keywords/vectors in the two baserun result sets ([SPE1CASE2](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareParallelSim_flow_SPE1CASE2/), [SPE9_CP_SHORT](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareParallelSim_flow_SPE9_CP_SHORT/), and [SPE3CASE1](https://ci.opm-project.org/job/opm-common-PR-builder/921/testReport/junit/(root)/mpi/compareParallelSim_flow_SPE3CASE1/))

Category two **appears to be** caused by comparing simulation results between sequential and parallel configurations (paths `${ROOT}/results/parallel/${TEST}/*.{INIT,EGRID,UNRST}` and `${ROOT}/tests/results/parallel/${TEST}/mpi/*.{INIT,EGRID,UNRST}`). Am I reading that correctly and, if so, is that the intended behaviour of the new result comparisons?

Category one is a little harder to pin down I think. Are the summary/SMSPEC tests supposed to compare all restart steps or just the final result?

**Comment by akva2** [comment]
jenkins build this opm-simulators=1827 please

**Comment by Tongdongq** [comment]
@bska I have a system with CentOS 7 and GCC 4.8.5. I appear to have problems with ecl as well, see https://github.com/OPM/opm-common/issues/799, does this error look familiar to you?

**Pull request merged**